options.go

```diff
@@ -0,0 +1,453 @@
+		return nil
+	}
+}
+
+// WithFlowSimpleCollections controls whether simple collections use flow style.
+//
+// When true, sequences and mappings containing only scalar values (no nested
+// collections) are rendered in flow style if they fit within the line width.
+// Example: {name: test, count: 42} or [a, b, c]
+// When called without arguments, defaults to true.
+//
+// When false (default), all collections use block style.
+func WithFlowSimpleCollections(flow ...bool) Option {
+	return func(o *options) error {
+		if len(flow) > 1 {
+			return errors.New("yaml: WithFlowSimpleCollections accepts at most one argument")
+		}
+		o.flowSimpleCollections = len(flow) == 0 || flow[0]
+		return nil
+	}
+}
+
+// Options combines multiple options into a single Option.
+// This is useful for creating option presets or combining version defaults
+// with custom options.
+//
+// Example:
+//
+//	opts := yaml.Options(v4.Opts(), yaml.WithIndent(3))
+//	yaml.Dump(&data, opts)
+func Options(opts ...Option) Option {
+	return func(o *options) error {
+		for _, opt := range opts {
+			if err := opt(o); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+// CombineOptions combines multiple options into a single Option.
+//
+// Deprecated: Use Options instead.
+func CombineOptions(opts ...Option) Option {
+	return Options(opts...)
+}
+
+// OptsYAML parses a YAML string containing option settings and returns
+// an Option that can be combined with other options using Options().
+//
+// The YAML string can specify any of these fields:
+//   - indent (int)
+//   - compact-seq-indent (bool)
+//   - line-width (int)
+//   - unicode (bool)
+//   - canonical (bool)
+//   - line-break (string: ln, cr, crln)
+//   - explicit-start (bool)
+//   - explicit-end (bool)
+//   - flow-simple-coll (bool)
+//   - known-fields (bool)
+//   - single-document (bool)
+//   - unique-keys (bool)
+//
+// Only fields specified in the YAML will override other options when
+// combined. Unspecified fields won't affect other options.
+//
+// Example:
+//
+//	opts, err := yaml.OptsYAML(`
+//	  indent: 3
+//	  known-fields: true
+//	`)
+//	yaml.Dump(&data, yaml.Options(V4, opts))
+func OptsYAML(yamlStr string) (Option, error) {
+	var cfg struct {
+		Indent                *int    `yaml:"indent"`
+		CompactSeqIndent      *bool   `yaml:"compact-seq-indent"`
+		LineWidth             *int    `yaml:"line-width"`
+		Unicode               *bool   `yaml:"unicode"`
+		Canonical             *bool   `yaml:"canonical"`
+		LineBreak             *string `yaml:"line-break"`
+		ExplicitStart         *bool   `yaml:"explicit-start"`
+		ExplicitEnd           *bool   `yaml:"explicit-end"`
+		FlowSimpleCollections *bool   `yaml:"flow-simple-coll"`
+		KnownFields           *bool   `yaml:"known-fields"`
+		SingleDocument        *bool   `yaml:"single-document"`
+		UniqueKeys            *bool   `yaml:"unique-keys"`
+	}
+	if err := Unmarshal([]byte(yamlStr), &cfg); err != nil {
+		return nil, err
+	}
```

Comment:
In such case I would use a map[string]string then loop on field name, this way I could report fields that are not supported.

This can be very useful to spot typos.

But the struct is pretty convenient, so maybe both can be done.

I don't think we already have an option yo report fields that are not mapped, right ?